assignment-1/IPL.java

```diff
@@ -0,0 +1,329 @@
+
+import static java.util.Map.Entry.comparingByValue;
+import static java.util.stream.Collectors.toMap;
+
+
+public class IPL {
+
+    static String[] teamNames = {"CSK", "MI", "RR", "RCB", "SRH", "PBKS", "DC", "KKR"};
+    static String[] venueNames = {"CHENNAI", "MUMBAI", "JAIPUR", "BENGALURU", "HYDERABAD", "CHANDIGARH", "DELHI", "KOLKATA"};
+    static ArrayList<Team> teams = new ArrayList<>();
+    static ArrayList<Match> matches = new ArrayList<>();
+
+    public static void fortyWicketsClub(String teamName) {
+        for (Team team : teams) {
+            if (team.teamName.equals(teamName)) {
+                for (Player p : team.squad) {
+                    if (p.wickets > 40) {
+                        p.display();
+                    }
+                }
+            }
+        }
+    }
+
+    public static void searchPlayersByName(String key) {
+        for (Team team : teams) {
+            for (Player p : team.squad) {
+                if (p.playerName.contains(key)) {
+                    p.display();
+                }
+            }
+
+        }
+    }
+
+    public static void getHighestWicketTakerOfTeam(String teamName) {
+        for (Team team : teams) {
+            if (team.teamName.equals(teamName)) {
+                Integer maxTillNow = 0;
+                Player bestBowler = null;
+                for (Player p : team.squad) {
+                    if (p.wickets >= maxTillNow) {
+                        bestBowler = p;
+                        maxTillNow = p.wickets;
+                    }
+                }
+                assert bestBowler != null;
+                bestBowler.display();
+            }
+
+        }
+    }
+
+    public static void getHighestRunScorerOfTeam(String teamName) {
+        for (Team team : teams) {
+            if (team.teamName.equals(teamName)) {
+                Integer maxTillNow = 0;
+                Player bestScorer = null;
+                for (Player p : team.squad) {
+                    if (p.runs >= maxTillNow) {
+                        bestScorer = p;
+                        maxTillNow = p.runs;
+                    }
+                }
+                assert bestScorer != null;
+                bestScorer.display();
+            }
+        }
+    }
+
+    public static void gettingTopPerformers() {
+        ArrayList<Player> top3Batsmen = getTopPlayersByRole("BATSMEN");
+        ArrayList<Player> top3Bowler = getTopPlayersByRole("BOWLER");
+        ArrayList<Player> top3AllRounder = getTopPlayersByRole("ALL ROUNDER");
+        displayArrayList(top3Batsmen);
+    }
+
+    static void displayArrayList(ArrayList<Player> players) {
+        for (Player player : players) {
+            player.display();
+        }
+    }
+
+    public static ArrayList<Player> getTopPlayersByRole(String role) {
+        Stack<Player> top3 = new Stack<>();
+        ArrayList<Player> result = new ArrayList<>();
+        for (Team team : teams) {
+            for (Player p : team.squad) {
```

Comment:
There should be a method in the Team class that returns top 3 batsmen, bowlers, and allrounders, and then you can find the top 3 in the entire IPL